driver/ppm_fillers.c

```diff
@@ -7978,3 +7978,81 @@ int f_sched_prog_fork(struct event_filler_arguments *args)
 	return add_sentinel(args);
 }
 #endif
+
+int f_sys_prctl_x(struct event_filler_arguments *args)
+{
+	int res;
+	int retval;
+	char *name = NULL;
+	syscall_arg_t option;
+	syscall_arg_t arg2;
+
+	/* Parameter 1: res (type: PT_ERRNO) */
+	retval = (int64_t)syscall_get_return_value(current, args->regs);
+	res = val_to_ring(args, retval, 0, false, 0);
+	CHECK_RES(res);
+
+	/*
+	 * option
+	 */
+	syscall_get_arguments_deprecated(current, args->regs, 0, 1, &option);
+	option = prctl_options_to_scap(option);
+	res = val_to_ring(args, option, 0, false, 0);
+	CHECK_RES(res);
+
+	/*
+	 * arg2
+	 */
+	syscall_get_arguments_deprecated(current, args->regs, 1, 1, &arg2);
+
+	switch(option){
+		case PPM_PR_GET_NAME:
+		case PPM_PR_SET_NAME:
+			/*
+			 * arg2_str
+			 */
+			if(likely(ppm_strncpy_from_user(args->str_storage, (const void __user *)arg2, PPM_MAX_PATH_SIZE) >= 0))
+			{
+				name = args->str_storage;
+				name[PPM_MAX_PATH_SIZE - 1] = '\0';
+			}
+			res = val_to_ring(args, (int64_t)(long)name, 0, false, 0);
```

Comment:
the `ppm_strncpy_from_user` stuff is already done inside the `val_to_ring`
```suggestion
			res = val_to_ring(args, arg2, 0, false, 0);
```

Reply:
here `from_user` has to be true 😅